### DISCUSSION TITLE
programmatically force color in non-tty env?

### DISCUSSION THREAD

**Discussion by moelholm** [google/zx#546] [Help needed]
With --color / FORCE_COLOR I can manually force colors etc even for non-tty circumstances.  Examples of how I can do it:

    FORCE_COLOR=1 tty-test.mjs > /dev/null # ✅

    TTY_TEST_OUTPUT=$(tty-test.mjs --color) # ✅

Is it possible to make this decision within the script? As in:

    tty-test.mjs > /dev/null  # ❌

    TTY_TEST_OUTPUT=$(tty-test.mjs) # ❌

Use case: I want to develop scripts that have conversations with the user on tty's ... while still allowing them to capture stdout. I can do this from within the code with some tty detection + stream proxying. Seems to work ok(ish) ...but then I also want to have colors and font formatting etc; that in turn requires forcing color support (somehow, either via args, env vars ... or possibly "programmatically" which is what this question/discussion is really about).

**Comment by zloirock** [comment]
I agree. I'm tired to install `FORCE_COLOR=1` in almost each script -)

**Comment by antonmedv** [comment]
Enable FORCE_COLOR=1 by default? It can break a lot of scripts.

**Reply by zloirock** [reply]
In case of a major release, now it does not looks like an insurmountable problem - still 2.6k dependent repos in the stat - it could be good to do before it became more popular. In addition, it can be good to add a simple way of configuring this, for example
```js
$.nocolors`command`;
$(command, { colors: false, somethingElse: true });
```
